src/rope/rope.rs

```diff
@@ -85,14 +87,40 @@ impl Rope {
     #[track_caller]
     #[inline]
     pub fn byte(&self, byte_index: usize) -> u8 {
+        self.get_byte(byte_index).unwrap_or_else(|| {
+            panic::line_index_out_of_bounds(byte_index, self.line_len())
+        })
+    }
+
+    /// Returns the byte at `byte_index`, if it exists.
+    ///
+    /// If `byte_index` is out of bounds, returns `None`.
+    ///
+    /// For a panic'ing version, see [`byte`](Self::byte).
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use crop::Rope;
+    /// #
+    /// let r = Rope::from("bar");
+    ///
+    /// assert_eq!(r.get_byte(0), Some(b'b'));
+    /// assert_eq!(r.get_byte(1), Some(b'a'));
+    /// assert_eq!(r.get_byte(2), Some(b'r'));
+    /// assert_eq!(r.get_byte(3), None);
+    /// ```
+    #[track_caller]
+    #[inline]
+    pub fn get_byte(&self, byte_index: usize) -> Option<u8> {
```

Comment:
I was going to duplicate these to `RopeSlice` as well, but I wondered, isn't it possible to have `impl Deref<Target = RopeSlice> for Rope`?